camera/src/main/java/com/kylecorry/andromeda/camera/Camera.kt

```diff
@@ -203,6 +214,16 @@ class Camera(
 
     }
 
+    override fun flipCamera() {
+        cameraSelector?.let {
+            cameraSelector = if (cameraSelector === CameraSelector.DEFAULT_FRONT_CAMERA)
+                CameraSelector.DEFAULT_BACK_CAMERA
+            else
+                CameraSelector.DEFAULT_FRONT_CAMERA
+            startImpl()
```

Comment:
Can this call stopImpl() first? There's some additional cleanup that I would like this to run since startImpl is only meant to be called once (until stopImpl is invoked). 

And also it should only call startImpl if it was already running

Reply:
I just tested this out, and it works for the most part but I noticed one bug:

If you do the following, you can get the camera to start up without a subscriber, which is not desired:
1. Start the camera
2. Stop the camera
3. Call flipCamera
4. The camera will start up again and stream to the preview but there are no active subscribers

Not sure if you are already aware of this, but if not can you add a check to see whether the camera is already on before calling startImpl again? You can either add a boolean to track whether it has been started, or you can just check if cameraProvider is not null (you would need to check prior to calling stopImpl).

Reply:
Thanks for the suggestion, i added a null safe scopefunction on cameraProvider, this should cover all cases where the camera is not activated.

Reply:
Thanks!